src/main/java/io/minio/Digest.java

```diff
@@ -233,27 +213,21 @@ public static String md5Hash(Object inputStream, int len)
 
       if (bytesRead < 0) {
         throw new InsufficientDataException("Insufficient data.  bytes read " + totalBytesRead + " expected "
-                                              + length);
+                + length);
       } else if (bytesRead == 0) {
         continue;
       }
 
-      sha256Digest.update(buf, 0, bytesRead);
-      md5Digest.update(buf, 0, bytesRead);
+      if(sha256Digest != null) {
+        sha256Digest.update(buf, 0, bytesRead);
+      }
+      if(md5Digest != null) {
+        md5Digest.update(buf, 0, bytesRead);
+      }
 
       totalBytesRead += bytesRead;
     } while (totalBytesRead < length);
-
-    if (file != null) {
-      file.seek(pos);
-    } else {
-      stream.reset();
-    }
-
-    String[] hashes = { BaseEncoding.base16().encode(sha256Digest.digest()).toLowerCase(),
-                        BaseEncoding.base64().encode(md5Digest.digest()) };
-
-    return hashes;
+    return pos;
   }
 
   private Digest() {}
```

Comment:
Please add a comment about this empty constructor.  Please leave two newlines between member functions.